process/transaction/transactionCostEstimator.go

```diff
@@ -186,6 +194,7 @@ func (tce *transactionCostEstimator) getTxGasLimit(tx *transaction.Transaction)
 
 	accountHandler, err := tce.accounts.LoadAccount(tx.SndAddr)
 	if err != nil || accountHandler == nil {
+		err = process.ErrInsufficientFunds
 		return
```

Comment:
in case this errors, (valid for all return branches) the function will output maxGasLimitPerBlock and a not nil error which does not follow the elrond-go pattern. Please remove output parameter naming and return default value (= 0 in this case) and no nil error